src/coreclr/vm/asyncthunks.cpp

```diff
@@ -202,37 +232,12 @@ void MethodDesc::EmitTaskReturningThunk(MethodDesc* pAsyncOtherVariant, MetaSig&
         // Catch
         {
             pCode->BeginCatchBlock(pCode->GetToken(CoreLibBinder::GetClass(CLASS__EXCEPTION)));
-
-            int fromExceptionToken;
-            if (logicalResultLocal != UINT_MAX)
-            {
-                MethodDesc* fromExceptionMD;
-                if (isValueTask)
-                    fromExceptionMD = CoreLibBinder::GetMethod(METHOD__VALUETASK__FROM_EXCEPTION_1);
-                else
-                    fromExceptionMD = CoreLibBinder::GetMethod(METHOD__TASK__FROM_EXCEPTION_1);
-
-                fromExceptionMD = FindOrCreateAssociatedMethodDesc(fromExceptionMD, fromExceptionMD->GetMethodTable(), FALSE, Instantiation(&thLogicalRetType, 1), FALSE);
-
-                fromExceptionToken = GetTokenForGenericMethodCallWithAsyncReturnType(pCode, fromExceptionMD);
-            }
-            else
-            {
-                MethodDesc* fromExceptionMD;
-                if (isValueTask)
-                    fromExceptionMD = CoreLibBinder::GetMethod(METHOD__VALUETASK__FROM_EXCEPTION);
-                else
-                    fromExceptionMD = CoreLibBinder::GetMethod(METHOD__TASK__FROM_EXCEPTION);
-
-                fromExceptionToken = pCode->GetToken(fromExceptionMD);
-            }
-            pCode->EmitCALL(fromExceptionToken, 1, 1);
-            pCode->EmitSTLOC(returnTaskLocal);
-            pCode->EmitLEAVE(returnTaskLabel);
```

Comment:
Can we keep `returnTaskLabel` and then change the catch handler to do something like `task = AsyncHelpers.FromException(ex)`?